components/sensor/max31865.rst

```diff
@@ -0,0 +1,70 @@
+MAX31865 Platinum RTD Temperature Sensor
+========================================
+
+.. seo::
+    :description: Instructions for setting up MAX31865 platinum RTD temperature sensors.
+    :image: max31865.jpg
+
+The ``max31865`` temperature sensor allows you to use your max31865 RTD
+temperature sensor (`datasheet <https://datasheets.maximintegrated.com/en/ds/MAX31865.pdf>`__) with ESPHome
+
+.. figure:: images/max31865-full.jpg
+    :align: center
+    :width: 50.0%
+
+    MAX31865 Sensor. Image by `Adafruit`_
+
+.. _Adafruit: https://www.adafruit.com/product/3328
+
+As the communication with the MAX31865 is done using SPI, you need
+to have an :ref:`spi bus <spi>` in your configuration with both **miso_pin** and **mosi_pin** set.
+
+- ``VIN`` connects to 5V (``3V3`` will output 3.3V), or directly connect ``3V3`` to 3.3V
+- ``GND`` connects to ground
+- ``CLK`` connects to the SPI **clk_pin**
+- ``SDO`` connects to the SPI **miso_pin**
+- ``SDI`` connects to the SPI **mosi_pin**
+- ``CS`` connects to a free GPIO pin
+- ``RDY`` is not used by this library
+
+.. code:: yaml
+
+    # Example configuration entry
+    spi:
+      miso_pin: D0
+      mosi_pin: D1
+      clk_pin: D2
+
+    sensor:
+      - platform: max31865
+        name: "Living Room Temperature"
+        cs_pin: D3
+        reference_resistance: 430 Ω
+        rtd_nominal_resistance: 100 Ω
+
+Configuration variables:
+------------------------
+
+- **name** (**Required**, string): The name for the temperature sensor.
+- **cs_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): The Chip Select pin of the SPI interface.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+- **reference_resistance** (**Required**, float): Reference resistor on the PCB. Adafruit's PT100 (#3328) uses 430 Ω, their PT1000 (#3648) uses 4300 Ω.
+- **rtd_nominal_resistance** (**Required**, float): Nominal resistance of the RTD at 0°C. PT100 is 100 Ω, PT1000 is 1000 Ω.
+- **filter** (*Optional*, string): The mains power frequency to reject (``50 Hz`` or ``60 Hz``). Defaults to ``60 Hz``.
```

Comment:
I think we should rename this option to `frequency_filter` or `mains_filter`.

Just `filter` is too close to `filters` used for sensors, and can be confusing.

Reply:
I like `mains_filter` as it helps avoid the implication that arbitrary frequencies can be filtered, and aligns with the datasheet specified intent of the filter:

> Noise from 50Hz or 60Hz power sources (including harmonics  of  the  ac  power’s  fundamental  frequency)  is  attenuated by 82dB.